### DISCUSSION TITLE
Bluetooth shell on xG24-DK2601B // Stack overflow when calling bt init

### DISCUSSION THREAD

**Discussion by Dermiste** [zephyrproject-rtos/zephyr#67733] [Q&A] [answered]
Hi all,

I am trying to build the shell sample for my [xG24-DK2601B](https://docs.zephyrproject.org/latest/boards/arm/efr32xg24_dk2601b/doc/index.html) board.

The build works fine, I can flash fine, also run a couple of commands in the shell such as help or other bt commands, but when I run `bt init` I get the following:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/1168706/5bd35f52-3d9b-4c2f-a2fc-149eeffb7763)

So far the only clue I found is that there is no mention of any bluetooth device in the device tree and various DTS ... could it be possible that the board definition is not complete? Other BT example seem to run fine, at least without stack overflow.

If anyone had a similar issue on a bluetooth board any help would be much appreciated.

@carlescufi or @andreeaDumitrache , any thoughts ?

**Comment by Dermiste** [comment]
Update, after looking into it I've put together a sort of call stack:
`subsys/bluetooth/shell/bt.c              line 1100:   err = bt_enable(bt_ready);`
`subsys/bluetooth/host/hci_core.c         line 4065:   err = bt_dev.drv->open();`
`drivers/bluetooth/hci/slz_hci.c          line  152:   ret = sl_btctrl_init_ll();`

I can't go any deeper, I assume it directly calls an already compiled driver afterwards.

Running an example such as `beacon `seems to work fine and execution then goes beyond  `sl_btctrl_init_ll()`, so I would assume it's more a context/thread issue than a driver issue.

**Comment by Dermiste** [comment]
Update, increasing CONFIG_SHELL_STACK_SIZE to 4096 instead of 2048 allowed execution to continue properly, new error is:
![image](https://github.com/zephyrproject-rtos/zephyr/assets/1168706/b0cee6ac-40ca-4854-ab68-37dfe2e45a9a)

A little digging shows that we are now stuck here:
`subsys/bluetooth/host/hci_core.c         line 4075: k_work_submit(&bt_dev.init);`

Deeper, some init function are executed:
`subsys/bluetooth/host/hci_core.c          line 3963: init_work()`

Up to this one which never returns:
`subsys/bluetooth/host/hci_core.c          line 3009: common_init()`

Hopefully blindly increasing stack sizes will help and we are not facing a much deeper issue such as a buffer not being emptied or sth like this.

**Comment by ndrs-pst** [comment]
@Dermiste, may need to increase `CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE` let's try to set to 2048.

By the way, the assert shown on indicate that assertion fail due to net_buf which seem link to this piece of code.

https://github.com/zephyrproject-rtos/zephyr/blob/3b99fb1b4a55ce51be185b25fd2ba99dcb4d0125/subsys/bluetooth/host/hci_core.c#L131-L133

**Reply by Dermiste** [reply]
Both BT_BUF_EVT_RX_SIZE and BT_BUF_CMD_TX_SIZE are equal to 255, it is also maximum value I believe

https://github.com/zephyrproject-rtos/zephyr/blob/3b99fb1b4a55ce51be185b25fd2ba99dcb4d0125/subsys/bluetooth/common/Kconfig#L111-L125

**Reply by ndrs-pst** [reply]
@Dermiste Another suggestion, please take a look in the following tests directory and seems that zephyr test it with nrf52840dk_nrf52840 board.

https://github.com/zephyrproject-rtos/zephyr/tree/main/tests/bluetooth/shell

**Reply by ndrs-pst** [reply]
@Dermiste, just to let you know, I got this .config file from doing the build in the `tests/bluetooth/shell` folder for `efr32xg24_dk2601b`. You might want to check it against yours to see if there's anything different.

[tests_bluetooth_shell_board_efr32xg24_dk2601b.zip](https://github.com/zephyrproject-rtos/zephyr/files/13979967/tests_bluetooth_shell_board_efr32xg24_dk2601b.zip)

**Reply by Dermiste** [reply]
Yes I figured all the test scenarios are designed for another board BUT I was definitely hoping they would work for mine, which is the main purpose of Zephyr I assume :)

Do you have the efr32xg24_dk2601b board available too? I assume the build output of yours vs mine would definitely be the same, unless you have tuned a working example.

After checking, yes .config files are the same.

**Comment by Dermiste** [comment, marked as answer]
Hi all, it's fixed, what did the trick for the first error:
`CONFIG_SHELL_STACK_SIZE=4096` 

Proposed fix:  in a ` efr32xg24_dk2601b.conf` in the sample folder OR here:
https://github.com/zephyrproject-rtos/zephyr/blob/3b99fb1b4a55ce51be185b25fd2ba99dcb4d0125/boards/arm/efr32xg24_dk2601b/Kconfig.defconfig#L28-L45

And for the second error:
`BT_HCI_RESERVE=1`

Proposed fix: 

Add a new case linked to BT_SILABS_HCI  with default 1 in:
https://github.com/zephyrproject-rtos/zephyr/blob/3b99fb1b4a55ce51be185b25fd2ba99dcb4d0125/subsys/bluetooth/host/Kconfig#L75-L93

**Comment by ndrs-pst** [comment]
@Dermiste, Good to hear that.
So according to MAINTANERS.yml I will tag @fkokosinski and @tgorochowik for the suggestion.

**Comment by tgorochowik** [comment]
@Dermiste could you please create a PR with your suggested fix? Please link it here afterwards, we will continue the discussion there.

**Reply by Dermiste** [reply]
https://github.com/zephyrproject-rtos/zephyr/pull/67991
